### PR TITLE
feat(fiber): add step for fiber migration

### PR DESCRIPTION
**Pull request by Steffy29**
ref: MANAGER-1365 and XDSL-5076

### CODE REVIEW

**Comment by Steffy29** [comment]
This PR is linked with https://github.com/ovh-ux/ovh-api-services/pull/106

File: client/components/telecom/pack/migration/pack-migration-process.service.js [hunk 1]
```diff
@@ -104,6 +104,14 @@ angular.module('managerApp').service('PackMigrationProcess', function ($q, OvhAp
       });
     }
 
+    // building details post params
+    postParams.buildingReference = migrationProcess.selectedOffer.buildingReference;
+    postParams.stair = migrationProcess.selectedOffer.stair;
+    postParams.floor = migrationProcess.selectedOffer.floor;
```
Comment (marie-j) [inline client/components/telecom/pack/migration/pack-migration-process.service.js]: You can also `_.assign(postParams, {
builidingReference: migrationOffer.selectedOffer.buildingReference,
stair: ...,
floor: ...
})`

File: client/components/telecom/pack/migration/pack-migration-process.service.js [hunk 2]
```diff
@@ -198,7 +207,9 @@ angular.module('managerApp').service('PackMigrationProcess', function ($q, OvhAp
 
   self.selectOffer = function (offer) {
     migrationProcess.selectedOffer = offer;
-    if (migrationProcess.selectedOffer.totalSubServiceToDelete > 0) {
+    if (migrationProcess.selectedOffer.offerName.toLowerCase().includes('ftth')) {
```
Comment (marie-j) [inline client/components/telecom/pack/migration/pack-migration-process.service.js]: prefer using `_.includes` as native js `includes` is not supported by all browsers

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+angular.module('managerApp').controller('TelecomPackMigrationBuildingDetailsCtrl', class TelecomPackMigrationBuildingDetailsCtrl {
+  constructor($q, $translate, PackMigrationProcess, OvhApiConnectivityEligibility) {
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js]: You can remove `$q` (unused).

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      engageMonths: null,
+      selectedBuilding: null,
+      selectedStair: null,
+      selectedFloor: null,
+    };
+
+    this.init();
+  }
+
+  init() {
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js]: Method body could be moved to `$onInit` method I guess.

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      this.OvhApiConnectivityEligibility.v6().buildingDetails({
+      }, params).$promise.then((buildingDetails) => {
+        if (_.has(buildingDetails, 'result.stairs')) {
+          this.process.selectedOffer.buildings[i].stairs = buildingDetails.result.stairs.map(
+            stair => this.convertStairs(stair),
+          );
+        }
+      });
+    });
+    this.loading.init = false;
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js]: Prefer `.finally`.

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    if (stair.floors[0] === '_NA_') {
+      stairsModel.floors = [{
+        label: this.$translate.instant('telecom_pack_migration_building_details_none'),
+        value: stair.floors[0],
+      }];
+    } else {
+      stairsModel.floors = [];
+      stair.floors.forEach((floor) => {
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.controller.js]: Prefer using `_.map` for that case.

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 7]
```diff
@@ -0,0 +1,69 @@
+<div class="telecom-pack-migration-shipping">
+    <div class="row">
+        <div class="col-xs-12 col-md-6">
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: -`col-xs-12`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            <div class="widget-presentation">
+                <!-- LOADER -->
+                <div class="text-center" data-ng-if="$ctrl.loading.init">
+                    <oui-spinner></oui-spinner>
+                </div>
+
+                <div data-ng-if="!$ctrl.loading.init" class="container">
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: Does `container` class is required?

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+                    <div>
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: This `div` could be removed if for instance you apply a `d-block` class directly on the `<span>`.

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+                    </div>
+
+                    <div>
+                        <span data-translate="telecom_pack_migration_building_details_select_building"></span>
+                        <oui-select name="building" data-model="$ctrl.model.selectedBuilding"
+                                    data-title="{{:: 'telecom_pack_migration_building_details_select_a_building' | translate}}"
+                                    placeholder="{{:: 'telecom_pack_migration_building_details_select_a_building' | translate}}"
+                                    data-items="$ctrl.process.selectedOffer.buildings" required data-match="name"
+                                    data-on-change="$ctrl.changeSelection()">
+                            <span ng-bind="$item.name"></span>
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: +`data-`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                        </oui-select>
+                    </div>
+
+                    <div ng-if="$ctrl.model.selectedBuilding != null">
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: +`data-`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+                        <span data-translate="telecom_pack_migration_building_details_select_stair"></span>
+                        <oui-select name="stair" data-model="$ctrl.model.selectedStair"
+                                    data-title="{{:: 'telecom_pack_migration_building_details_select_a_stair' | translate}}"
+                                    placeholder="{{:: 'telecom_pack_migration_building_details_select_a_stair' | translate}}"
+                                    data-items="$ctrl.model.selectedBuilding.stairs" data-align="start" data-match="stair.label"
+                                    data-on-change="$ctrl.changeSelection(true)">
+                            <span ng-bind="$item.stair.label"></span>
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: +`data-`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+                        </oui-select>
+                    </div>
+
+                    <div ng-if="$ctrl.model.selectedStair != null">
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: +`data-`

Comment (antleblanc) [reply client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: `!==`?

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+                        <span data-translate="telecom_pack_migration_building_details_select_floor"></span>
+                        <oui-select name="floor" data- model="$ctrl.model.selectedFloor"
+                                    data-title="{{:: 'telecom_pack_migration_building_details_select_a_floor' | translate}}"
+                                    placeholder="{{:: 'telecom_pack_migration_building_details_select_a_floor' | translate}}"
+                                    data-items="$ctrl.model.selectedStair.floors" data-align="start" data-match="label">
+                            <span ng-bind="$item.label"></span>
```
Comment (antleblanc) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: +`data-`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html (hunk 8 above)
Comment (Steffy29) [reply client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: not really

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 15]
```diff
@@ -1,21 +1,19 @@
 <div class="telecom-pack-migration-shipping">
     <div class="row">
-        <div class="col-xs-12 col-md-6">
+        <div class="-col-xs-12">
```
Comment (FredericEspiau) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: @antleblanc meant `class="col-md-6"`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 16]
```diff
@@ -1,21 +1,19 @@
...
-                    <div>
-                        <span data-translate="telecom_pack_migration_building_details_select_engagment"></span>
-                        <oui-radio name="engagement"
-                                   data-ng-repeat="engageMonths in $ctrl.process.selectedOffer.engagementMonths track by $index"
-                                   data-value="engageMonths" model="$ctrl.model.engageMonths">{{engageMonths}} {{
-                            'telecom_pack_migration_building_details_engagment_month' | translate }}
-                        </oui-radio>
-                    </div>
+                <div data-ng-if="!$ctrl.loading.init">
+                    <span class="d-block" data-translate="telecom_pack_migration_building_details_select_engagment"></span>
```
Comment (FredericEspiau) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: `span.d-block` => `p`

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html (hunk 15 above)
Comment (Steffy29) [reply client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: it wasn't clear

Comment (FredericEspiau) [reply client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: he hardly is

File: client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html [hunk 17]
```diff
@@ -1,60 +1,56 @@
...
-                    <button type="button"
-                            class="btn btn-primary"
-                            data-ng-disabled="!$ctrl.isValidSelection()"
-                            data-translate="telecom_pack_migration_building_details_confirm"
-                            data-ng-click="$ctrl.nextStep()">
-                    </button>
-                </div>
+            <div data-ng-if="$ctrl.model.selectedStair !== null">
+                <span data-translate="telecom_pack_migration_building_details_select_floor"></span>
+                <oui-select name="floor" data- model="$ctrl.model.selectedFloor"
```
Comment (FredericEspiau) [inline client/app/telecom/pack/migration/building-details/telecom-pack-migration-building-details.html]: Extra space betwwen `data-` and `model`

**Comment by ovh-cds** [comment]
CDS Report build-manager#214.1 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✔ 
* Stage 3
  * Prepare tar.gz for UXCI / Release ✘

**Comment by ovh-cds** [comment]
CDS Report build-manager#216.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✔ 
* Stage 3
  * Prepare tar.gz for UXCI / Release ✘

**Comment by ovh-cds** [comment]
CDS Report build-manager#216.1 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✔ 
  * Build manager US ✔ 
  * Tests ✔ 
* Stage 3
  * Prepare tar.gz for UXCI / Release ✘

**Pull request merged**